pkg/util/namespace_test.go

```diff
@@ -0,0 +1,264 @@
+package util
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	coretesting "k8s.io/client-go/testing"
+)
+
+func TestIsNamespaceExist(t *testing.T) {
+	type args struct {
+		client    kubernetes.Interface
+		namespace string
+		reactor   coretesting.ReactionFunc
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "query namespace error",
+			args: args{
+				client:    fake.NewSimpleClientset(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}),
+				namespace: "default",
+				reactor: func(action coretesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &corev1.Namespace{}, errors.New("failed to get namespace")
+				},
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "namespace not exists",
+			args: args{
+				client:    fake.NewSimpleClientset(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default-1"}}),
+				namespace: "default",
+				reactor:   nil,
+			},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name: "namespace already exists",
+			args: args{
+				client:    fake.NewSimpleClientset(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}}),
+				namespace: "default",
+				reactor:   nil,
+			},
+			want:    true,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.args.reactor != nil {
+				tt.args.client.(*fake.Clientset).PrependReactor("get", "namespaces", tt.args.reactor)
+			}
+
+			got, err := IsNamespaceExist(tt.args.client, tt.args.namespace)
+			if (err != nil) != tt.wantErr {
```

Comment:
```suggestion
			if (err == nil && tt.wantErr == true) || (err != nil && tt.wantErr == false) {
```
Suggestion for readability.

Reply:
Ok, it's modified, thanks for pointing it out